src/dataExplorer/context/tags.tsx

```diff
@@ -82,6 +91,53 @@ export const TagsProvider: FC<Prop> = ({children, scope}) => {
 
     setLoadingTagKeys(RemoteDataState.Loading)
 
+    if (isFlagEnabled('v2privateQueryUI') && isIOx) {
+      // user input is sanitized to avoid SQL injection
+      const sanitized = sanitizeSQLSearchTerm(searchTerm)
+      const queryTextSQL: string = `
+        SELECT column_name
+        FROM information_schema.columns
+        WHERE
+            table_schema = 'iox'
+            AND data_type LIKE 'Dictionary%'
+            AND table_name = '${measurement}'
+            AND column_name ILIKE '%${sanitized}%'
+        LIMIT ${DEFAULT_LIMIT}
+      `
+      const newTags: Tags = {}
+      try {
+        const resp = await queryAPI(queryTextSQL, scope, {
+          language: LanguageType.SQL,
+          bucket,
+        } as QueryOptions)
+        const keys = (resp.parsed.table.columns?.column_name?.data ??
+          []) as string[]
+
+        // Initialize tags with keys
+        keys.map(key => {
+          newTags[key] = []
+        })
+
+        // Initialize status for each key
+        const tagValueStatuses = {} as Hash<RemoteDataState>
+        keys.map(key => {
+          tagValueStatuses[key] = RemoteDataState.NotStarted
+        })
+
```

Comment:
Could we avoid a second map but initializing both the tags and statuses in a single iteration?

const tagValueStatuses = {}

`keys.map(key => {
   newTags[key] = []
   tagValueStatuses[key] = RemoteDataState.NotStarted
})

Reply:
Of course yes! Updated in https://github.com/influxdata/ui/pull/6791/commits/26ec5315ae39fc461447cea6b182a42953fef6c0